sysutils/pfSense-pkg-apcupsd/files/usr/local/www/widgets/widgets/apcupsd.widget.php

```diff
@@ -0,0 +1,407 @@
+			$user_settings["widgets"][$widgetkey]["apc_charge_warning_threshold"] = "50";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_charge_critical_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_charge_critical_threshold"] = "15";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_bage_warning_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_bage_warning_threshold"] = "365";
+		}
+		if (!isset($user_settings["widgets"][$widgetkey]["apc_bage_critical_threshold"])) {
+			$user_settings["widgets"][$widgetkey]["apc_bage_critical_threshold"] = "720";
+		}
+		
+		if (!(include_once "/usr/local/pkg/apcupsd.inc")) {
+			$rtnstr = "";
+			$rtnstr .= "<div class=\"panel panel-warning responsive\"><div class=\"panel-heading\"><h2 class=\"panel-title\">apcupsd not installed...</h2></div>\n";
+			$rtnstr .= "<pre>\n";
+			$rtnstr .= "Please install and configure apcupsd before using this widget.<br />\n";
+			$rtnstr .= "</pre>\n";
+			$rtnstr .= "</div>\n";
+			print($rtnstr);
+			exit(1);
+		}
+		
+		if (check_nis_running_apcupsd()) {
+			$nisip = (check_nis_ip_apcupsd() != ''? check_nis_ip_apcupsd() : "localhost");
+			$nisport = (check_nis_port_apcupsd() != '' ? check_nis_port_apcupsd() : "3551");
+
+			$ph = popen("/usr/local/sbin/apcaccess -h " . escapeshellarg($nisip) . ":" . escapeshellarg($nisport) . " 2>&1", "r" );
+			while ($v = fgets($ph)) {
+				$results[trim(explode(': ',$v)[0])]=trim(explode(': ',$v)[1]);
+			}
+			pclose($ph);
+
+			$rtnstr = "";
+
+			$rtnstr .= "<tr><td>Status</td><td colspan=\"3\">\n";
+			
+			if($results != null) {
+				switch ($results['STATUS']) {
+					default:
+					case 'ONBATT':
+						if (str_replace(" Percent", "", $results['BCHARGE']) <= 25) {
+							$rtnstr .= "<span class=\"fa fa-battery-empty\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 50) {
+							$rtnstr .= "<span class=\"fa fa-battery-quarter\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 75) {
+							$rtnstr .= "<span class=\"fa fa-battery-half\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 99) {
+							$rtnstr .= "<span class=\"fa fa-battery-three-quarters\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						} else {
+							$rtnstr .= "<span class=\"fa fa-battery-full\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						}
```

Comment:
This block could be simplified if the value of `str_replace(" Percent", "", $results['BCHARGE'])` is stored on a variable and then the items that change on each line are defined.  Something like:

```php
$percent = str_replace(" Percent", "", $results['BCHARGE']);

if ($percent <= 25) {
    $class = "fa-battery-empty";
} elseif ($percent <= 50) {
    $class = "fa-battery-quarter";
...
}

$rtnstr .= "<span class=\"fa {$class}\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
```

Also note I used `elseif` instead of `else if` because it's the desired style in pfSense code